src/demos/zenodo/Results.js

```diff
@@ -0,0 +1,156 @@
+      break;
+    case 'right':
+      style = { marginRight: size };
+      break;
+    default:
+      style = { margin: `0 ${size}` };
+  }
+  return <span style={style}>{text}</span>;
+};
+
+export class Results extends Component {
+  constructor(props) {
+    super(props);
+
+    this.sortValues = this.props.sortValues;
+    this.resultsPerPageValues = this.props.resultsPerPageValues;
+  }
+
+  renderResultsListItem = (result, index) => {
+    const metadata = result.metadata;
+    return (
+      <Item key={index} href={`#${metadata.recid}`}>
+        <Item.Image
+          size="small"
+          src={result.imageSrc || 'http://placehold.it/200'}
+        />
+        <Item.Content>
+          <Item.Header>{metadata.title}</Item.Header>
+          <Item.Description>
+            {_truncate(metadata.description, { length: 200 })}
+          </Item.Description>
+        </Item.Content>
+      </Item>
+    );
+  };
+
+  renderResultsGridItem(result, index) {
+    const metadata = result.metadata;
+    return (
+      <Card fluid key={index} href={`#${metadata.recid}`}>
+        <Image src={result.imageSrc || 'http://placehold.it/200'} />
+        <Card.Content>
+          <Card.Header>{metadata.title}</Card.Header>
+          <Card.Description>
+            {_truncate(metadata.description, { length: 200 })}
+          </Card.Description>
+        </Card.Content>
+      </Card>
+    );
+  }
+
+  render() {
+    const { total } = this.props.currentResultsState.data;
+    const CustomResultsListCmp = () => (
+      <ResultsList renderListItem={this.renderResultsListItem} />
+    );
+    const CustomResultsGridCmp = () => (
+      <ResultsGrid renderGridItem={this.renderResultsGridItem} />
+    );
+    return total ? (
+      <div>
+        <Grid relaxed>
+          <ActiveAggregations
+            renderSelectedAggregation={renderSelectedAggregation}
+          />
+        </Grid>
+        <Grid relaxed verticalAlign="middle">
+          <Grid.Column width={8}>
+            <SpanWithMargin text="Found" margin="right" />
+            <Count />
+            <SpanWithMargin text="results sorted by" />
+            <Sort values={this.sortValues} />
+          </Grid.Column>
+          <Grid.Column width={8} textAlign="right">
+            <SpanWithMargin text="Show" margin="right" />
+            <ResultsPerPage
+              values={this.resultsPerPageValues}
+              defaultValue={10}
+            />
+            <SpanWithMargin text="results per page" />
+            <LayoutSwitcher defaultLayout="grid" />
+          </Grid.Column>
+        </Grid>
+        <Grid relaxed style={{ padding: '2em 0' }}>
+          <ResultsMultiLayout
+            resultsListCmp={CustomResultsListCmp}
+            resultsGridCmp={CustomResultsGridCmp}
+          />
+        </Grid>
+        <Grid relaxed verticalAlign="middle" textAlign="center">
+          <Pagination />
+        </Grid>
+      </div>
+    ) : null;
+  }
+}
+
+Results.propTypes = {};
```

Comment:
Shouldn't `sortValues` and `resultsPerPageValues` be a part of that?

Reply:
Yes they should, but this is a demo example, not really the lib, I am not sure it is so important :)